### PR TITLE
Update octokit authentication mechanism

### PR DESCRIPTION
**Pull request by confusdcodr**

### CODE REVIEW

File: handler.js [hunk 1]
```diff
@@ -237,122 +236,103 @@ module.exports.build_done = (event, context, callback) => {
         target_url: 'https://' + region + '.console.aws.amazon.com/codebuild/home?region=' + region + '#/builds/' + event.build.id + '/view/new',
         context: githubContext,
         description: 'Build ' + buildStatus + '...'
-      }).catch(function(err){
-        console.log(err);
-        context.fail(data);
```
Comment (lorengordon) [inline handler.js:242]: where did this one go?

File: handler.js [hunk 2]
```diff
@@ -136,98 +132,101 @@ module.exports.start_build = (event, context, callback) => {
...
 
-  setGithubAuth(octokit, ssm, ssmParams, function (err) {
+  console.log('Github state will be', state)
+
+  setGithubAuth(ssm, ssmParams, function (err, octokit) {
     if (err) {
-      console.log(err);
-      callback(err);
+      console.log(err)
+      callback(err)
```
Comment (lorengordon) [inline handler.js:240]: i'm not super sure, but does the callback need to include two values (now that `octokit` is in the callback definition)
```suggestion
      callback(err, null)
```

Comment (lorengordon) [reply handler.js:240]: Oh, sorry, in the wrong function for that. derp.

Comment (confusdcodr) [reply handler.js:240]: Given the other code that has multiple returns in their callback, I do not think so.

File: handler.js (hunk 1 above)
Comment (confusdcodr) [reply handler.js:242]: I intentionally removed it as `data` is undefined.

File: handler.js [hunk 3]
```diff
@@ -237,122 +236,103 @@ module.exports.build_done = (event, context, callback) => {
...
+  // attempt to get credentials
+  ssm.getParameter(params.accessToken, function (err, data) {
+    if (err) {
+      callback(err)
+    } else {
+      try {
+        octokit = new Octokit({
+          auth: data.Parameter.Value
+        })
+        callback(null, octokit)
```
Comment (lorengordon) [inline handler.js]: any particular reason to move the callback into the try/catch? seems like that could "catch" some kind recursive exception...

File: handler.js (hunk 1 above)
Comment (confusdcodr) [reply handler.js:242]: Should I put it back as `context.fail(err)`

File: handler.js [hunk 4]
```diff
@@ -237,122 +236,103 @@ module.exports.build_done = (event, context, callback) => {
...
+      } catch (err) {
+        callback(err)
       }
-    });
-  }
+    }
+  })
 }
 
-/*
```
Comment (lorengordon) [inline handler.js:281]: why remove the docstring?

File: handler.js (hunk 3 above)
Comment (confusdcodr) [reply handler.js]: Not really. Was just being overly cautious

File: handler.js (hunk 4 above)
Comment (confusdcodr) [reply handler.js:281]: because I wasn't paying attention to what it was and thought it was just commented out code

File: handler.js (hunk 1 above)
Comment (lorengordon) [reply handler.js:242]: actually, as far as i can tell, in current versions of the aws sdk for javascript, the `context` object no longer has a `fail()` method. so probably needs to be removed throughout, in favor of handling with `callback(err)`...

Comment (confusdcodr) [reply handler.js:242]: Putting back as `context.fail(err)`

Comment (lorengordon) [reply handler.js:242]: https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html#nodejs-handler-sync

Comment (confusdcodr) [reply handler.js:242]: Ok, I think this is fixed.

File: handler.js [hunk 5]
```diff
@@ -237,122 +247,99 @@ module.exports.build_done = (event, context, callback) => {
         target_url: 'https://' + region + '.console.aws.amazon.com/codebuild/home?region=' + region + '#/builds/' + event.build.id + '/view/new',
         context: githubContext,
         description: 'Build ' + buildStatus + '...'
-      }).catch(function(err){
-        console.log(err);
-        context.fail(data);
-      });
+      }).catch(function (err) {
+        console.log(err)
```
Comment (lorengordon) [inline handler.js:251]: i think we should update this with the correct call to signal the err here... yeah, realize this wasn't exactly working before, but it _was_ exiting just not the right way... :/
```suggestion
        console.log(err)
        callback(err)
```

Comment (lorengordon) [reply handler.js:251]: @confusdcodr just this one thing, then good to go

Comment (confusdcodr) [reply handler.js:251]: Fixed

**Pull request merged**